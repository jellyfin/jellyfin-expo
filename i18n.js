/**
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
import * as Localization from 'expo-localization';
import i18next from 'i18next';
import { initReactI18next } from 'react-i18next';
import { I18nManager } from 'react-native';

import ar from './langs/ar.json';
import be from './langs/be.json';
import bg from './langs/bg.json';
import ca from './langs/ca.json';
import cs from './langs/cs.json';
import cy from './langs/cy.json';
import da from './langs/da.json';
import de from './langs/de.json';
import el from './langs/el.json';
import en from './langs/en.json';
import eo from './langs/eo.json';
import es from './langs/es.json';
import es_419 from './langs/es_419.json';
import es_AR from './langs/es_AR.json';
import et from './langs/et.json';
import fa from './langs/fa.json';
import fi from './langs/fi.json';
import fil from './langs/fil.json';
import fr from './langs/fr.json';
import ga from './langs/ga.json';
import he from './langs/he.json';
import hr from './langs/hr.json';
import hu from './langs/hu.json';
import hy from './langs/hy.json';
import id from './langs/id.json';
import is from './langs/is.json';
import it from './langs/it.json';
import ja from './langs/ja.json';
import kk from './langs/kk.json';
import ko from './langs/ko.json';
import lt from './langs/lt.json';
import lv from './langs/lv.json';
import mn from './langs/mn.json';
import ms from './langs/ms.json';
import mt from './langs/mt.json';
import nb_NO from './langs/nb_NO.json';
import nds from './langs/nds.json';
import nl from './langs/nl.json';
import pl from './langs/pl.json';
import pt from './langs/pt.json';
import pt_BR from './langs/pt_BR.json';
import pt_PT from './langs/pt_PT.json';
import ro from './langs/ro.json';
import ru from './langs/ru.json';
import sk from './langs/sk.json';
import sl from './langs/sl.json';
import sq from './langs/sq.json';
import sr from './langs/sr.json';
import sv from './langs/sv.json';
import ta from './langs/ta.json';
import tr from './langs/tr.json';
import uk from './langs/uk.json';
import vi from './langs/vi.json';
import zh_Hans from './langs/zh_Hans.json';
import zh_Hant from './langs/zh_Hant.json';

export const resources = {
	en: { translation: en },
	ar: { translation: ar },
	be: { translation: be },
	bg: { translation: bg },
	ca: { translation: ca },
	cs: { translation: cs },
	cy: { translation: cy },
	da: { translation: da },
	de: { translation: de },
	el: { translation: el },
	eo: { translation: eo },
	es: { translation: es },
	'es-419': { translation: es_419 },
	'es-AR': { translation: es_AR },
	et: { translation: et },
	fa: { translation: fa },
	fi: { translation: fi },
	fil: { translation: fil },
	fr: { translation: fr },
	ga: { translation: ga },
	he: { translation: he },
	hr: { translation: hr },
	hu: { translation: hu },
	hy: { translation: hy },
	id: { translation: id },
	is: { translation: is },
	it: { translation: it },
	ja: { translation: ja },
	kk: { translation: kk },
	ko: { translation: ko },
	lt: { translation: lt },
	lv: { translation: lv },
	mn: { translation: mn },
	ms: { translation: ms },
	mt: { translation: mt },
	'nb-NO': { translation: nb_NO },
	nds: { translation: nds },
	nl: { translation: nl },
	pl: { translation: pl },
	pt: { translation: pt },
	'pt-BR': { translation: pt_BR },
	'pt-PT': { translation: pt_PT },
	ro: { translation: ro },
	ru: { translation: ru },
	sk: { translation: sk },
	sl: { translation: sl },
	sq: { translation: sq },
	sr: { translation: sr },
	sv: { translation: sv },
	ta: { translation: ta },
	tr: { translation: tr },
	uk: { translation: uk },
	vi: { translation: vi },
	'zh-Hans': { translation: zh_Hans },
	'zh-Hant': { translation: zh_Hant }
};

// Ensure RTL layout is enabled for RTL locales
I18nManager.forceRTL(Localization.isRTL);

i18next
	.use(initReactI18next)
	.init({
		// debug: true,
		fallbackLng: 'en',
		// Use the fallback language for empty strings
		returnEmptyString: false,
		lng: Localization.locale,
		interpolation: {
			escapeValue: false
		},
		resources
	});

// Export i18next instance for use in tests
export default i18next;
