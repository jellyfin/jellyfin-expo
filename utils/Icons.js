/**
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
import { Platform } from 'react-native';

export const getIconName = (name = '') => {
	if (name) {
		return name
		// return Platform.OS === 'ios' ? `ios-${name}` : `md-${name}`;
	}
	return name;
};
