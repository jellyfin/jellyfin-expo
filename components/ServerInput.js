/**
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
import { useNavigation, useNavigationState } from '@react-navigation/native';
import PropTypes from 'prop-types';
import React, { useContext, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { ActivityIndicator, Platform, StyleSheet } from 'react-native';
import { Input, ThemeContext } from 'react-native-elements';

import Screens from '../constants/Screens';
import { useStores } from '../hooks/useStores';
import { getIconName } from '../utils/Icons';
import { parseUrl, validateServer } from '../utils/ServerValidator';

const sanitizeHost = (url = '') => url.trim();

// FIXME: eslint fails to parse the propTypes properly here
const ServerInput = React.forwardRef(({
	onError = () => { /* noop */ }, // eslint-disable-line react/prop-types
	onSuccess = () => { /* noop */ }, // eslint-disable-line react/prop-types
	...props
}, ref) => {
	const [ host, setHost ] = useState('');
	const [ isValidating, setIsValidating ] = useState(false);
	const [ isValid, setIsValid ] = useState(true);
	const [ validationMessage, setValidationMessage ] = useState('');

	const { serverStore, settingStore } = useStores();
	const navigation = useNavigation();
	const navigationState = useNavigationState((state) => state);
	const { t } = useTranslation();
	const { theme } = useContext(ThemeContext);

	const onAddServer = async () => {
		console.log('add server', host);
		if (!host) {
			setIsValid(false);
			setValidationMessage(t('addServer.validation.empty'));
			onError();
			return;
		}

		setIsValidating(true);
		setIsValid(true);
		setValidationMessage('');

		// Parse the entered url
		let url;
		try {
			url = parseUrl(host);
			console.log('parsed url', url);
		} catch (err) {
			console.info(err);
			setIsValidating(false);
			setIsValid(false);
			setValidationMessage(t('addServer.validation.invalid'));
			onError();
			return;
		}

		// Validate the server is available
		const validation = await validateServer({ url });
		console.log(`Server is ${validation.isValid ? '' : 'not '}valid`);
		if (!validation.isValid) {
			const message = validation.message || 'invalid';
			setIsValidating(false);
			setIsValid(validation.isValid);
			setValidationMessage(t([ `addServer.validation.${message}`, 'addServer.validation.invalid' ]));
			onError();
			return;
		}

		// Save the server details
		serverStore.addServer({ url });
		settingStore.set({ activeServer: serverStore.servers.length - 1 });
		// Call the success callback
		onSuccess();

		// Navigate to the main screen
		navigation.replace(
			Screens.MainScreen,
			{
				screen: Screens.HomeTab,
				params: {
					screen: Screens.HomeScreen,
					params: { activeServer: settingStore.activeServer }
				}
			}
		);
	};

	return (
		<Input
			testID='server-input'
			ref={ref}
			inputContainerStyle={{
				...styles.inputContainerStyle,
				backgroundColor: theme.colors.searchBg
			}}
			leftIcon={{
				name: getIconName('globe-outline'),
				type: 'ionicon',
				color: theme.colors.grey3
			}}
			leftIconContainerStyle={styles.leftIconContainerStyle}
			labelStyle={{
				color: theme.colors.grey1
			}}
			placeholderTextColor={theme.colors.grey3}
			rightIcon={isValidating ? <ActivityIndicator /> : null}
			selectionColor={theme.colors.primary}
			autoCapitalize='none'
			autoCorrect={false}
			autoCompleteType='off'
			autoFocus={true}
			keyboardType={Platform.OS === 'ios' ? 'url' : 'default'}
			returnKeyType='go'
			textContentType='URL'
			editable={!isValidating}
			value={host}
			errorMessage={isValid ? null : validationMessage}
			onChangeText={text => setHost(sanitizeHost(text))}
			onSubmitEditing={() => onAddServer()}
			{...props}
		/>
	);
});

ServerInput.propTypes = {
	onError: PropTypes.func,
	onSuccess: PropTypes.func
};

const styles = StyleSheet.create({
	inputContainerStyle: {
		marginTop: 8,
		marginBottom: 12,
		borderRadius: 3,
		borderBottomWidth: 0
	},
	leftIconContainerStyle: {
		marginLeft: 12
	}
});

export default ServerInput;
