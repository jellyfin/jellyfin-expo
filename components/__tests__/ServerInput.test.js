/**
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

import { NavigationContainer } from '@react-navigation/native';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import React from 'react';

import { parseUrl, validateServer } from '../../utils/ServerValidator';
import ServerInput from '../ServerInput';

import '../../i18n';

jest.mock('../../utils/ServerValidator');
parseUrl.mockImplementation(url => url);
validateServer.mockResolvedValue({ isValid: true });

describe('ServerInput', () => {
	beforeEach(() => { 
		// jest.resetModules(); // this causes `Cannot read properties of null (reading 'useRef')`
		jest.clearAllMocks();
	});

	it('should render correctly', async () => {
		const { toJSON } = render( // Cannot read properties of 'null' (reading 'useRef')
			<NavigationContainer>
				<ServerInput />
			</NavigationContainer>
		);

		expect(toJSON()).toMatchSnapshot();
	});

	it('should show error when input is blank', async () => {
		const onError = jest.fn();
		const onSuccess = jest.fn();
		const { getByTestId, toJSON } = render(
			<NavigationContainer>
				<ServerInput
					onError={onError}
					onSuccess={onSuccess}
				/>
			</NavigationContainer>
		);
		const input = getByTestId('server-input');

		fireEvent(input, 'onSubmitEditing');

		await waitFor(() => expect(onError).toHaveBeenCalled());
		expect(onSuccess).not.toHaveBeenCalled();
		expect(parseUrl).not.toHaveBeenCalled();
		expect(validateServer).not.toHaveBeenCalled();

		expect(toJSON()).toMatchSnapshot();
	});

	it('should show error when url is undefined', async () => {
		const onError = jest.fn();
		const onSuccess = jest.fn();
		const { getByTestId, toJSON } = render(
			<NavigationContainer>
				<ServerInput
					onError={onError}
					onSuccess={onSuccess}
				/>
			</NavigationContainer>
		);
		const input = getByTestId('server-input');

		fireEvent(input, 'onChangeText', undefined);
		fireEvent(input, 'onSubmitEditing');

		await waitFor(() => expect(onError).toHaveBeenCalled());
		expect(onSuccess).not.toHaveBeenCalled();
		expect(parseUrl).not.toHaveBeenCalled();
		expect(validateServer).not.toHaveBeenCalled();

		expect(toJSON()).toMatchSnapshot();
	});

	it('should show error when url is whitespace', async () => {
		const onError = jest.fn();
		const onSuccess = jest.fn();
		const { getByTestId, toJSON } = render(
			<NavigationContainer>
				<ServerInput
					onError={onError}
					onSuccess={onSuccess}
				/>
			</NavigationContainer>
		);
		const input = getByTestId('server-input');

		fireEvent(input, 'onChangeText', '   	');
		fireEvent(input, 'onSubmitEditing');

		await waitFor(() => expect(onError).toHaveBeenCalled());
		expect(onSuccess).not.toHaveBeenCalled();
		expect(parseUrl).not.toHaveBeenCalled();
		expect(validateServer).not.toHaveBeenCalled();

		expect(toJSON()).toMatchSnapshot();
	});

	it.skip('should succeed for valid urls', async () => { // navigation.replace is not a function
		const onError = jest.fn();
		const onSuccess = jest.fn();
		const { getByTestId, toJSON } = render(
			<NavigationContainer>
				<ServerInput
					onError={onError}
					onSuccess={onSuccess}
				/>
			</NavigationContainer>
		);
		const input = getByTestId('server-input');
		fireEvent(input, 'onChangeText', 'test');
		fireEvent(input, 'onSubmitEditing');

		await waitFor(() => expect(onSuccess).toHaveBeenCalled());
		expect(onError).not.toHaveBeenCalled();
		expect(parseUrl).toHaveBeenCalled();
		expect(validateServer).toHaveBeenCalled();

		expect(toJSON()).toMatchSnapshot();
	});

	it('should show error if parseUrl throws', async () => {
		const onError = jest.fn();
		const onSuccess = jest.fn();
		const { getByTestId, toJSON } = render(
			<NavigationContainer>
				<ServerInput
					onError={onError}
					onSuccess={onSuccess}
				/>
			</NavigationContainer>
		);

		parseUrl.mockImplementationOnce(() => {
			throw new Error('test error');
		});

		const input = getByTestId('server-input');
		fireEvent(input, 'onChangeText', 'test');
		fireEvent(input, 'onSubmitEditing');

		await waitFor(() => expect(onError).toHaveBeenCalled());
		expect(onSuccess).not.toHaveBeenCalled();
		expect(parseUrl).toHaveBeenCalled();
		expect(validateServer).not.toHaveBeenCalled();

		expect(toJSON()).toMatchSnapshot();
	});

	it('should show error if url is invalid', async () => {
		const onError = jest.fn();
		const onSuccess = jest.fn();
		const { getByTestId, toJSON } = render(
			<NavigationContainer>
				<ServerInput
					onError={onError}
					onSuccess={onSuccess}
				/>
			</NavigationContainer>
		);

		validateServer.mockResolvedValueOnce({ isValid: false });

		const input = getByTestId('server-input');
		fireEvent(input, 'onChangeText', 'test');
		fireEvent(input, 'onSubmitEditing');

		await waitFor(() => expect(onError).toHaveBeenCalled());
		expect(onSuccess).not.toHaveBeenCalled();
		expect(parseUrl).toHaveBeenCalled();
		expect(validateServer).toHaveBeenCalled();

		expect(toJSON()).toMatchSnapshot();
	});
});
