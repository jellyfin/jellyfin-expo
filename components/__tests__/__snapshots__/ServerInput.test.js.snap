// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerInput should render correctly 1`] = `
<View
  style={
    {
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#303337",
        "borderBottomWidth": 0,
        "borderColor": "#86939e",
        "borderRadius": 3,
        "flexDirection": "row",
        "marginBottom": 12,
        "marginTop": 8,
        "transform": [
          {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 12,
          "marginVertical": 4,
          "paddingRight": 4,
        }
      }
    >
      <View
        style={
          {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
    <TextInput
      autoCapitalize="none"
      autoCompleteType="off"
      autoCorrect={false}
      autoFocus={true}
      editable={true}
      keyboardType="url"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#86939e"
      returnKeyType="go"
      selectionColor="#2089dc"
      style={
        {
          "color": "#242424",
          "flex": 1,
          "fontSize": 18,
          "minHeight": 40,
        }
      }
      testID="server-input"
      textContentType="URL"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <Text
    style={
      {
        "color": "#ff190c",
        "fontSize": 12,
        "margin": 5,
      }
    }
  />
</View>
`;

exports[`ServerInput should show error if parseUrl throws 1`] = `
<View
  style={
    {
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#303337",
        "borderBottomWidth": 0,
        "borderColor": "#86939e",
        "borderRadius": 3,
        "flexDirection": "row",
        "marginBottom": 12,
        "marginTop": 8,
        "transform": [
          {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 12,
          "marginVertical": 4,
          "paddingRight": 4,
        }
      }
    >
      <View
        style={
          {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#86939e",
                    "fontSize": 24,
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                  {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <TextInput
      autoCapitalize="none"
      autoCompleteType="off"
      autoCorrect={false}
      autoFocus={true}
      editable={true}
      keyboardType="url"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#86939e"
      returnKeyType="go"
      selectionColor="#2089dc"
      style={
        {
          "color": "#242424",
          "flex": 1,
          "fontSize": 18,
          "minHeight": 40,
        }
      }
      testID="server-input"
      textContentType="URL"
      underlineColorAndroid="transparent"
      value="test"
    />
  </View>
  <Text
    style={
      {
        "color": "#ff190c",
        "fontSize": 12,
        "margin": 5,
      }
    }
  >
    Server Address is invalid
  </Text>
</View>
`;

exports[`ServerInput should show error if url is invalid 1`] = `
<View
  style={
    {
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#303337",
        "borderBottomWidth": 0,
        "borderColor": "#86939e",
        "borderRadius": 3,
        "flexDirection": "row",
        "marginBottom": 12,
        "marginTop": 8,
        "transform": [
          {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 12,
          "marginVertical": 4,
          "paddingRight": 4,
        }
      }
    >
      <View
        style={
          {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#86939e",
                    "fontSize": 24,
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                  {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <TextInput
      autoCapitalize="none"
      autoCompleteType="off"
      autoCorrect={false}
      autoFocus={true}
      editable={true}
      keyboardType="url"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#86939e"
      returnKeyType="go"
      selectionColor="#2089dc"
      style={
        {
          "color": "#242424",
          "flex": 1,
          "fontSize": 18,
          "minHeight": 40,
        }
      }
      testID="server-input"
      textContentType="URL"
      underlineColorAndroid="transparent"
      value="test"
    />
  </View>
  <Text
    style={
      {
        "color": "#ff190c",
        "fontSize": 12,
        "margin": 5,
      }
    }
  >
    Server Address is invalid
  </Text>
</View>
`;

exports[`ServerInput should show error when input is blank 1`] = `
<View
  style={
    {
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#303337",
        "borderBottomWidth": 0,
        "borderColor": "#86939e",
        "borderRadius": 3,
        "flexDirection": "row",
        "marginBottom": 12,
        "marginTop": 8,
        "transform": [
          {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 12,
          "marginVertical": 4,
          "paddingRight": 4,
        }
      }
    >
      <View
        style={
          {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#86939e",
                    "fontSize": 24,
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                  {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <TextInput
      autoCapitalize="none"
      autoCompleteType="off"
      autoCorrect={false}
      autoFocus={true}
      editable={true}
      keyboardType="url"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#86939e"
      returnKeyType="go"
      selectionColor="#2089dc"
      style={
        {
          "color": "#242424",
          "flex": 1,
          "fontSize": 18,
          "minHeight": 40,
        }
      }
      testID="server-input"
      textContentType="URL"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <Text
    style={
      {
        "color": "#ff190c",
        "fontSize": 12,
        "margin": 5,
      }
    }
  >
    Server Address cannot be empty
  </Text>
</View>
`;

exports[`ServerInput should show error when url is undefined 1`] = `
<View
  style={
    {
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#303337",
        "borderBottomWidth": 0,
        "borderColor": "#86939e",
        "borderRadius": 3,
        "flexDirection": "row",
        "marginBottom": 12,
        "marginTop": 8,
        "transform": [
          {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 12,
          "marginVertical": 4,
          "paddingRight": 4,
        }
      }
    >
      <View
        style={
          {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#86939e",
                    "fontSize": 24,
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                  {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <TextInput
      autoCapitalize="none"
      autoCompleteType="off"
      autoCorrect={false}
      autoFocus={true}
      editable={true}
      keyboardType="url"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#86939e"
      returnKeyType="go"
      selectionColor="#2089dc"
      style={
        {
          "color": "#242424",
          "flex": 1,
          "fontSize": 18,
          "minHeight": 40,
        }
      }
      testID="server-input"
      textContentType="URL"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <Text
    style={
      {
        "color": "#ff190c",
        "fontSize": 12,
        "margin": 5,
      }
    }
  >
    Server Address cannot be empty
  </Text>
</View>
`;

exports[`ServerInput should show error when url is whitespace 1`] = `
<View
  style={
    {
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#303337",
        "borderBottomWidth": 0,
        "borderColor": "#86939e",
        "borderRadius": 3,
        "flexDirection": "row",
        "marginBottom": 12,
        "marginTop": 8,
        "transform": [
          {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 12,
          "marginVertical": 4,
          "paddingRight": 4,
        }
      }
    >
      <View
        style={
          {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#86939e",
                    "fontSize": 24,
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                  {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <TextInput
      autoCapitalize="none"
      autoCompleteType="off"
      autoCorrect={false}
      autoFocus={true}
      editable={true}
      keyboardType="url"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#86939e"
      returnKeyType="go"
      selectionColor="#2089dc"
      style={
        {
          "color": "#242424",
          "flex": 1,
          "fontSize": 18,
          "minHeight": 40,
        }
      }
      testID="server-input"
      textContentType="URL"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <Text
    style={
      {
        "color": "#ff190c",
        "fontSize": 12,
        "margin": 5,
      }
    }
  >
    Server Address cannot be empty
  </Text>
</View>
`;

exports[`ServerInput should succeed for valid urls 1`] = `
<View
  style={
    {
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#303337",
        "borderBottomWidth": 0,
        "borderColor": "#86939e",
        "borderRadius": 3,
        "flexDirection": "row",
        "marginBottom": 12,
        "marginTop": 8,
        "transform": [
          {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "marginLeft": 12,
          "marginVertical": 4,
          "paddingRight": 4,
        }
      }
    >
      <View
        style={
          {
            "overflow": "hidden",
          }
        }
      >
        <View>
          <View
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                [
                  {
                    "color": "#86939e",
                    "fontSize": 24,
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                  {
                    "fontFamily": "ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  {},
                ]
              }
              testID="iconIcon"
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <TextInput
      autoCapitalize="none"
      autoCompleteType="off"
      autoCorrect={false}
      autoFocus={true}
      editable={false}
      keyboardType="url"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#86939e"
      returnKeyType="go"
      selectionColor="#2089dc"
      style={
        {
          "color": "#242424",
          "flex": 1,
          "fontSize": 18,
          "minHeight": 40,
        }
      }
      testID="server-input"
      textContentType="URL"
      underlineColorAndroid="transparent"
      value="test"
    />
    <View
      style={
        {
          "alignItems": "center",
          "height": 40,
          "justifyContent": "center",
          "marginVertical": 4,
          "paddingRight": 4,
        }
      }
    >
      <ActivityIndicator />
    </View>
  </View>
  <Text
    style={
      {
        "color": "#ff190c",
        "fontSize": 12,
        "margin": 5,
      }
    }
  />
</View>
`;
